crates/torii/core/src/query_queue.rs

```diff
@@ -97,6 +97,45 @@ impl QueryQueue {
                     let broker_message = BrokerMessage::EntityUpdated(entity_updated);
                     self.push_publish(broker_message);
                 }
+                QueryType::DeleteEntity(entity) => {
+                    let delete_model = query.execute(&mut *tx).await.with_context(|| {
+                        format!("Failed to execute query: {:?}, args: {:?}", statement, arguments)
+                    })?;
+                    if delete_model.rows_affected() == 0 {
+                        continue;
+                    }
+
+                    let row = sqlx::query(
+                        "UPDATE entities SET updated_at=CURRENT_TIMESTAMP, executed_at=?, \
+                         event_id=? WHERE id = ? RETURNING *",
+                    )
+                    .bind(entity.block_timestamp)
+                    .bind(entity.event_id)
+                    .bind(entity.entity_id)
+                    .fetch_one(&mut *tx)
+                    .await?;
+                    let mut entity_updated = EntityUpdated::from_row(&row)?;
+                    entity_updated.updated_model = Some(entity.entity);
+
+                    let count = sqlx::query_scalar::<_, i64>(
+                        "SELECT count(*) FROM entity_model WHERE entity_id = ?",
+                    )
+                    .bind(entity_updated.id.clone())
+                    .fetch_one(&mut *tx)
+                    .await?;
```

Comment:
To have the code organized around that we should move (in the future, not this PR) to some functions that could make this less verbose and easier to re-use, especially counting or generic stuff like so.